### PR TITLE
Add smoothing to flashlight movement

### LINKED ISSUE #4665: osu! flashlight should not follow the mouse 1:1
In stable it has its own movement easing/interpolation, which give it a nice smoothing effect when following the mouse. In lazer it is currently locked hard to the mouse.

### PR DESCRIPTION
**Pull request by holly-hacker**
Smooths the movement of the Flashlight texture the same way osu-stable does.

Closes #4665.

### CODE REVIEW

**Comment by HoLLy-HaCKeR** [comment]
Right now this only applies to osu!, not Catch. Should I apply it there too?

File: osu.Game.Rulesets.Osu/Mods/OsuModFlashlight.cs [hunk 1]
```diff
@@ -48,7 +52,16 @@ public void OnSliderTrackingChange(ValueChangedEvent<bool> e)
...
+                var position = FlashlightPosition;
+                var destination = e.MousePosition;
+
+                double frameTime = Clock.CurrentTime - lastPositionUpdate;
+                double interp = Interpolation.ApplyEasing(Easing.Out, MathHelper.Clamp(frameTime / follow_delay, 0, 1));
+                FlashlightPosition = new Vector2(
+                    (float)Interpolation.Lerp(position.X, destination.X, interp),
+                    (float)Interpolation.Lerp(position.Y, destination.Y, interp)
+                );
+                lastPositionUpdate = Clock.CurrentTime;
```
Comment (smoogipoo) [inline osu.Game.Rulesets.Osu/Mods/OsuModFlashlight.cs]: I believe this is all equivalent to:

```
FlashlightPosition = Interpolation.ValueAt(MathHelper.Clamp(Clock.ElapsedFrameTime, 0, follow_delay), position, destination, 0, follow_delay, Easing.Out)
```

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game.Rulesets.Osu/Mods/OsuModFlashlight.cs (hunk 1 above)
Comment (HoLLy-HaCKeR) [reply osu.Game.Rulesets.Osu/Mods/OsuModFlashlight.cs]: Not sure how I missed ValueAt, changed it now.

**Review by smoogipoo — DISMISSED** [review]
lgtm

**Review by peppy — APPROVED** [review]

**Pull request merged**